lib/transport/selenium-webdriver/method-mappings.js

```diff
@@ -332,6 +332,19 @@ module.exports = class MethodMappings {
         async switchToFrame(frameId) {
           if (frameId === undefined) {
             frameId = null;
+          } else if (typeof frameId === 'string'){
+            try {
```

Comment:
I'm afraid this code is a bit too weird to be understandable and also it is almost duplicating the line containing `this.driver.findElement`. Also we shouldn't overwrite the `frameId` parameter.

Reply:
In order to avoid the nested try/catch you can simply use a .catch() on the first promise, e.g.:
```js
frameId = await this.driver.findElement(By.id(frameId)).catch(err => {
  return this.driver.findElement(By.name(frameId));
})
```